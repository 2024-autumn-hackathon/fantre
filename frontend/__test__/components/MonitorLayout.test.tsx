import { render } from "@testing-library/react"

import LayoutView from "@/components/MonitorLayout"

describe("Monitor Layout Snapshot", () => {
  const layoutView = render(
    <LayoutView
      headerContent="Header"
      mainContent="Main"
      navigationContent="Navi"
      footerContent="Footer"
    />
  ).container

  it("layout snapshot", () => {
    expect(layoutView).toMatchSnapshot()
  })
})
