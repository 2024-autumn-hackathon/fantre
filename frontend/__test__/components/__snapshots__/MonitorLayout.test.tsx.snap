// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitor Layout Snapshot layout snapshot 1`] = `
<div>
  <section
    class="grid relative w-full h-full rounded-[50px] bg-my-light-blue grid-rows-[40px_1fr_40px] grid-cols-[20px_1fr_20px] X-mob:grid-cols-[50px_5fr_20px_4fr_50px] X-tab:h-[calc(100%-50px)] X-tab:grid-rows-[40px_1fr_40px] X-tab:grid-cols-[20px_5fr_54fr_20px_26fr_6fr_20px]"
  >
    <section
      class="flex items-center justify-end col-start-2 col-end-3 X-tab:col-start-3 X-tab:col-end-6 X-mob:col-start-4 X-mob:col-end-5"
    >
      Header
    </section>
    <section
      class="h-full max-h-[calc(100vh-80px)] row-start-2 row-end-3 col-start-2 col-end-3 X-tab:max-h-[calc(100vh-130px)] X-mob:col-end-5 X-tab:col-end-7 grid grid-rows-[5fr_20px_4fr] grid-cols-1 X-mob:grid-rows-1 X-mob:grid-cols-[5fr_20px_4fr] X-tab:grid-rows-1 X-tab:grid-cols-[59fr_20px_32fr]"
    >
      <section
        class="h-full bg-my-gray rounded-3xl relative flex flex-col justify-around  overflow-auto"
      >
        Main
      </section>
      <section
        class="h-full flex flex-col justify-around items-center bg-my-gray rounded-3xl overflow-auto row-start-3 X-mob:row-start-1 X-mob:row-end-2 X-mob:col-start-3 X-tab:row-start-1 X-tab:col-start-3"
      >
        Navi
      </section>
    </section>
    <section
      class="row-start-3 col-start-2 col-end-3 X-tab:col-start-3 X-tab:row-start-3 X-tab:col-end-6 X-mob:row-start-1 X-mob:row-end-2 X-mob:col-start-2 X-mob:col-end-3"
    >
      Footer
    </section>
    <img
      alt="電源ボタンの見た目を表現する画像です"
      class="hidden X-tab:block row-start-3 col-start-6 col-end-7"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      src="/PowerButtonImage.svg"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </section>
  <div
    class="relative w-full h-[50px] hidden X-tab:block"
  >
    <img
      alt="モニターの台座・コード・びりびりマークの見た目を表現する画像です"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      src="/MonitorDecorationsImage.svg"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</div>
`;
